### PR TITLE
fix `Hammerhead emit an error "Failed to find a DNS-record for the resource"` (close #1384)

### LINKED ISSUE #1384: Hammerhead emit an error "Failed to find a DNS-record for the resource"
https://testcafe-discuss.devexpress.com/t/failed-to-find-a-dns-record-for-the-resource-at/574

Server for reproducing:
```js
const net = require('net');

let mainPageSocket = null;

const server = net.createServer(socket => {
    socket.on('data', data => {
        const stringData = data.toString();
        const url = stringData.match(/GET ([\s\S]+) HTTP/)[1];
        const isKeepAlive = /Connection: keep-alive/gi.test(stringData);

        if (url === '/') {
            mainPageSocket = socket;

            socket.setKeepAlive(isKeepAlive);
            socket.write([
                'HTTP/1.1 302 Object Moved',
                'Location: /redirect',
                'Content-Length: 0',
                '',
                ''
            ].join('\r\n'));
        } else if (url === '/redirect') {
            mainPageSocket && mainPageSocket.end();

            if (mainPageSocket !== socket) {
                socket.write([
                    'HTTP/1.1 200 Ok',
                    'Content-Length: 5',
                    '',
                    'Hello'
                ].join('\r\n'));
            }
        }
    });
}).listen(8124, () => console.log('server bound'));
```
The error occurs when I try to open `http://localhost:8124/` through hammerhead.
* The hammerhead creates connection for `http://localhost:8124/` with the `keep-alive` header.
* Server responds with the `302` redirect and `keep alive` connection.
* The hammerhead uses a created connection for `http://localhost:8124/redirect`.
* Server closes the connection.
* The hammerhead socket emits the `Error: socket hang up` error.
* The hammerhead emits the `Failed to find a DNS-record for the resource` error

Hammerhead logs:
```
>>>>>>>>>>>>>>>> request created <<<<<<<<<<<<<<<<
http://localhost:8124
>>>>>>>>>>>>>>>> lookup <<<<<<<<<<<<<<<<
http://localhost:8124
>>>>>>>>>>>>>>>> connect <<<<<<<<<<<<<<<<
http://localhost:8124
>>>>>>>>>>>>>>>> data <<<<<<<<<<<<<<<<
http://localhost:8124
>>>>>>>>>>>>>>>> request created <<<<<<<<<<<<<<<<
http://localhost:8124/redirect
>>>>>>>>>>>>>>>> end <<<<<<<<<<<<<<<<
http://localhost:8124
{ Error: socket hang up
    at createHangUpError (_http_client.js:330:15)
    at Socket.socketOnEnd (_http_client.js:423:23)
    at emitNone (events.js:120:20)
    at Socket.emit (events.js:218:7)
    at endReadableNT (_stream_readable.js:1054:12)
    at _combinedTickCallback (internal/process/next_tick.js:138:11)
    at process._tickDomainCallback (internal/process/next_tick.js:218:9) code: 'ECONNRESET' }
Failed to find a DNS-record for the resource at <a href="http://localhost:8124/redirect">http://localhost:8124/redirect</a>.
>>>>>>>>>>>>>>>> end <<<<<<<<<<<<<<<<
http://localhost:8124/redirect
```

### PR DESCRIPTION
**Pull request by LavrovArtem**
/cc @miherlosev @churkin

### CODE REVIEW

**Comment by LavrovArtem** [comment]
@testcafe-build-bot \retest

**Comment by LavrovArtem** [comment]
@testcafe-build-bot \retest

File: src/request-pipeline/destination-request/index.js [hunk 1]
```diff
@@ -125,6 +126,9 @@ export default class DestinationRequest extends EventEmitter {
     }
 
     _onError (err) {
+        if (err.message && SOCKET_HANG_UP_ERR_RE.test(err.message))
```
Comment (miherlosev) [inline src/request-pipeline/destination-request/index.js]: 1)Move  `err.message && SOCKET_HANG_UP_ERR_RE.test(err.message)` to `_isSocketHungUpErr(err)` function

2)Move to on module level regular expressions from body of  the`_isDNSErr` function

**Comment by LavrovArtem** [comment]
@testcafe-build-bot \retest

**Comment by LavrovArtem** [comment]
@testcafe-build-bot \retest

**Pull request merged**